src/layer/__tests__/retries.ts

```diff
@@ -0,0 +1,83 @@
+import 'jest-setup';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+
+import { constructFixtureFindTiles } from './retries.fixtures';
+
+const mockNetwork = new MockAdapter(axios);
+
+test('Retries correctly on network errors', async () => {
+  // we need to adjust jest timeout until we have support for setting the delay when retrying,
+  // otherwise the test will time out:
+  jest.setTimeout(7000);
+  const {
+    fromTime,
+    toTime,
+    bbox,
+    layer,
+    mockedResponse,
+    expectedResultTiles,
+    expectedResultHasMore,
+  } = constructFixtureFindTiles({});
+
+  mockNetwork.reset();
+  mockNetwork.onPost().replyOnce(500);
+  mockNetwork.onPost().replyOnce(500);
+  mockNetwork.onPost().replyOnce(200, mockedResponse);
+
+  const { tiles, hasMore } = await layer.findTiles(bbox, fromTime, toTime);
+
+  expect(mockNetwork.history.post.length).toBe(3);
+  expect(tiles).toStrictEqual(expectedResultTiles);
+  expect(hasMore).toBe(expectedResultHasMore);
+});
+
+test("Doesn't retry if not needed", async () => {
+  const {
+    fromTime,
+    toTime,
+    bbox,
+    layer,
+    mockedResponse,
+    expectedResultTiles,
+    expectedResultHasMore,
+  } = constructFixtureFindTiles({});
+
+  mockNetwork.reset();
+  mockNetwork.onPost().replyOnce(200, mockedResponse);
+
+  const { tiles, hasMore } = await layer.findTiles(bbox, fromTime, toTime);
+
+  expect(mockNetwork.history.post.length).toBe(1);
+  expect(tiles).toStrictEqual(expectedResultTiles);
+  expect(hasMore).toBe(expectedResultHasMore);
+});
+
+test('Fails if max. retries (default == 2) was exceeded', async () => {
+  jest.setTimeout(2 * 3000 + 1000);
+  const { fromTime, toTime, bbox, layer } = constructFixtureFindTiles({});
+
+  mockNetwork.reset();
+  mockNetwork.onPost().replyOnce(500);
+  mockNetwork.onPost().replyOnce(500);
+  mockNetwork.onPost().replyOnce(500);
+
+  await expect(layer.findTiles(bbox, fromTime, toTime)).rejects.toThrow();
+
+  expect(mockNetwork.history.post.length).toBe(3);
+});
+
+test('Fails if max. retries (explicitly set) was exceeded', async () => {
+  jest.setTimeout(3 * 3000 + 1000);
+  const { fromTime, toTime, bbox, layer } = constructFixtureFindTiles({});
+
+  mockNetwork.reset();
+  mockNetwork.onPost().replyOnce(500);
+  mockNetwork.onPost().replyOnce(500);
+  mockNetwork.onPost().replyOnce(500);
+  mockNetwork.onPost().replyOnce(500);
+
+  await expect(layer.findTiles(bbox, fromTime, toTime, null, null, { retries: 3 })).rejects.toThrow();
+
+  expect(mockNetwork.history.post.length).toBe(4);
+});
```

Comment:
I wanted to add  test for disabled retrying. According to documentation `The default of `retries` is `null` (disabled).` I would expect that making request without  RequestConfiguration  or with  RequestConfiguration {retries:null} or {retries:0} would result in single request being made, but this is not the case.  

```
test("Doesn't retry if retrying is disabled", async () => {
  jest.setTimeout(3 * 3000 + 1000);

  const { fromTime, toTime, bbox, layer } = constructFixtureFindTiles({});
  mockNetwork.reset();
  mockNetwork.onPost().replyOnce(500);
  mockNetwork.onPost().replyOnce(500);
  mockNetwork.onPost().replyOnce(500);

  await expect(layer.findTiles(bbox, fromTime, toTime, null, null, { retries: null })).rejects.toThrow();
  expect(mockNetwork.history.post.length).toBe(1);
});
```

Reply:
Great catch, thank you!

There was a mismatch between the docs and the implementation, and the implementation didn't work anyway (no way to disable retries). I have now:
- set default value to `2` (sentinelhub-py has a default value for `max_download_attempts` set to 4, but that seemed a bit high to me for our use-case - thank you @zcernigoj for the idea of checking there!),
- made `null` a "use default" value, similar to how it is used with parameters
- fixed the documentation and tests to reflect that.

Does this change make sense or do you think some other values would be more appropriate?